src/changelog/.2.x.x/2515_datetimeformatter_formatto.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<entry xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns="https://logging.apache.org/xml/ns"
+       xsi:schemaLocation="https://logging.apache.org/xml/ns https://logging.apache.org/xml/ns/log4j-changelog-0.xsd"
+       type="changed">
+  <issue id="2515" link="https://github.com/apache/logging-log4j2/pull/2515"/>
+  <description format="asciidoc">Use DateTimeFormatter.formatTo to directly append to StringBuilder</description>
```

Comment:
I have massaged the text to make it look more user-friendly in the release notes:
```suggestion
  <description format="asciidoc">Replace some usages of `DateTimeFormatter#toString()` with `DateTimeFormatter#formatTo(StringBuilder)` to cut down on allocations</description>
```